libvast/include/vast/system/spawn_arguments.hpp

```diff
@@ -56,17 +56,19 @@ struct spawn_arguments {
 };
 
 /// Attempts to parse `query` as ::expression.
-caf::expected<expression> parse_expression(const std::string& query);
+caf::expected<std::pair<expression, std::optional<pipeline>>>
+parse_query(const std::string& query);
 
 /// Attempts to parse `[begin, end)` as ::expression.
-caf::expected<expression>
-parse_expression(std::vector<std::string>::const_iterator begin,
-                 std::vector<std::string>::const_iterator end);
+caf::expected<std::pair<expression, std::optional<pipeline>>>
+parse_query(std::vector<std::string>::const_iterator begin,
+            std::vector<std::string>::const_iterator end);
 
-caf::expected<expression>
-parse_expression(const std::vector<std::string>& args);
+caf::expected<std::pair<expression, std::optional<pipeline>>>
+parse_query(const std::vector<std::string>& args);
 
-caf::expected<expression> parse_expression(const spawn_arguments& args);
+caf::expected<std::pair<expression, std::optional<pipeline>>>
+parse_query(const spawn_arguments& args);
```

Comment:
This really does not belong into `spawn_arguments.hpp`. After all, we just removed eager expression parsing from the spawn arguments.

I would suggest putting this into a new set of functions called `vast/parse_query.{hpp,cpp}`.

Reply:
I moved the `parse_query` functions into new files `vast/system/parse_query.{hpp,cpp}`